### PR TITLE
Port-MurmurHash

### PR DESCRIPTION
**Pull request by MakoEnergy**

### CODE REVIEW

File: include/MurmurHash.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    };//MurmurHash128_x86
+
+    /// @brief A convenience type for the return value of a x64 128-bit Hashing function.
+    struct MurmurHashResult_128_x64
+    {
+        /// @brief Array of 64-bit unsigned integers forming the 128-bit hash.
+        UInt64 Hash[2] = {};
+    };//MurmurHash128_x64
+
+    /// @brief A Murmur hash function for 32-bit platforms that produces a 32-bit hash.
```
Comment (Sqeaky) [inline include/MurmurHash.h]: What does this mean running it on AMD64?

File: include/MurmurHash.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    /// @param Key A pointer to the data that will be hashed.
+    /// @param Length The byte length of the Key.
+    /// @param Seed The base number to scramble the hash with.
```
Comment (Sqeaky) [inline include/MurmurHash.h]: Would call this a salt be more cryptographically accurate?

File: include/MurmurHash.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    /// @return Returns a 128-bit Hash of the Key.
+    MurmurHashResult_128_x86 MurmurHash3_x86_128(const void* Key, const SizeType Length, const UInt32 Seed);
+    /// @brief A Murmur hash function for 64-bit platforms that produces a 128-bit hash.
+    /// @param Key A pointer to the data that will be hashed.
+    /// @param Length The byte length of the Key.
+    /// @param Seed The base number to scramble the hash with.
+    /// @return Returns a 128-bit Hash of the Key.
+    MurmurHashResult_128_x64 MurmurHash3_x64_128(const void* Key, const SizeType Length, const UInt32 Seed);
```
Comment (Sqeaky) [inline include/MurmurHash.h]: The bit size and arch keep swapping left and right sides, can this be made more uniform?

File: include/MurmurHash.h [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+   John Blackwood - makoenergy02@gmail.com
+*/
+//-----------------------------------------------------------------------------
+// MurmurHash3 was written by Austin Appleby, and is placed in the public
+// domain. The author hereby disclaims copyright to this source code.
+
+// Note - The x86 and x64 versions do _not_ produce the same results, as the
+// algorithms are optimized for their respective platforms. You can still
+// compile and run any of them on any platform, but your performance with the
+// non-native version will be less than optimal.
```
Comment (Sqeaky) [inline include/MurmurHash.h]: This should be in the docs, could you make a small @page for this?

File: src/MurmurHash.cpp [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+
+namespace {
+    using namespace Mezzanine;
+    //-----------------------------------------------------------------------------
+    // Platform-specific functions and macros
+
+    #if defined(_MSC_VER) // Microsoft Visual Studio
+        #define FORCE_INLINE __forceinline
+
+        #include <stdlib.h>
```
Comment (Sqeaky) [inline src/MurmurHash.cpp]: Include inside an anonymous namespace, that is eyebrow raising, but It works and it targets a single compiler.

File: include/MurmurHash.h (hunk 1 above)
Comment (MakoEnergy) [reply include/MurmurHash.h]: You might have worse performance relative to the other methods, maybe.  I've updated the docs.

File: include/MurmurHash.h (hunk 2 above)
Comment (MakoEnergy) [reply include/MurmurHash.h]: Fixed.

File: include/MurmurHash.h (hunk 4 above)
Comment (MakoEnergy) [reply include/MurmurHash.h]: It's lazily tacked on, but there.

File: include/MurmurHash.h (hunk 3 above)
Comment (MakoEnergy) [reply include/MurmurHash.h]: Fixed.

**Pull request merged**